File: gonit-korbo-joi_tamim-shahriar-subeen/chapter-04/02_divisor_efficiently.py
while True:
    n = input ("Please enter a positive integer. Enter 0 to quit: ")
    n = int(n)
    if n == 0:
        break

    if n < 0:
        print("You must enter a positive integer. Please try again.")
        continue
    
    factors = [1, n]

    for i in range(2, (n//2) + 1): # (n / 2) + 1
        if n % i == 0:
            factors.append(i)
    
    factors = sorted(factors)
    print("Factors of", n, ":", factors)
    print("\n")

File: chapter-04/02_divisor_efficiently.c
#include <stdio.h> 

int main() {
    int num, i;

    while(1) {
        printf("Please enter a positive integer. Enter 0 to quit\n: ");
        scanf("%d", &num);
        if (num == 0) {
            break;
        }
        if (num < 0) {
            printf("You must enter a positive integer. Please try again.\n");
            continue;
        }
        printf("Factors of %d : 1", num);
        for (i = 2; i <= (num/2)+1; i++) {
            if (num % i == 0) {
                printf(" %d", i);
            }
        }
        printf(" %d\n\n", num);
    }

    return 0;
}